src/vs/workbench/services/configurationResolver/browser/baseConfigurationResolverService.ts

```diff
@@ -346,9 +349,9 @@ export abstract class BaseConfigurationResolverService extends AbstractVariableR
 						}
 					}
 					const pickOptions: IPickOptions<PickStringItem> = { placeHolder: info.description, matchOnDetail: true, ignoreFocusLost: true };
-					return this.quickInputService.pick(picks, pickOptions, undefined).then(resolvedInput => {
+					return this.userInputAccessQueue.queue(() => this.quickInputService.pick(picks, pickOptions, undefined)).then(resolvedInput => {
```

Comment:
The `this.commandService.executeCommand` on line 364 should also be queued since a command can trigger quick input.